src/jit/lowerxarch.cpp

```diff
@@ -2571,39 +2572,56 @@ void Lowering::TreeNodeInfoInitSIMD(GenTree* tree)
                 info->srcCount = 1;
             }
 
-            // If the index is not a constant, we will use the SIMD temp location to store the vector.
-            // Otherwise, if the baseType is floating point, the targetReg will be a xmm reg and we
-            // can use that in the process of extracting the element.
-            //
-            // If the index is a constant and base type is a small int we can use pextrw, but on AVX
-            // we will need a temp if are indexing into the upper half of the AVX register.
-            // In all other cases with constant index, we need a temp xmm register to extract the
-            // element if index is other than zero.
-
-            if (!op2->IsCnsIntOrI())
+            GenTree* op1 = tree->gtOp.gtOp1;
+            if (op1->isMemoryOp())
             {
-                (void)comp->getSIMDInitTempVarNum();
+                MakeSrcContained(tree, op1);
+
+                // GT_IND of TYP_SIMD12 also reserves an internal float
+                // register for reading 4 and 8 bytes from memory and
+                // assembling them into target XMM reg.  In this case
+                // we don't need internal registers.
```

Comment:
This comment is confusing because the code here isn't doing anything special with `TYP_SIMD12`. I might reword it as "Although GT_IND of TYP_SIMD12 reserves ..., that is not required in this case even for TYP_SIMD12, because we are loading a single element." or something like that.